experiments/hello-network-socket-ping/local_ping.c

```diff
@@ -0,0 +1,175 @@
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+#include <sys/un.h>
+#include <sys/socket.h>
+
+#include "shared.h"
+#include "util.h"
+
+#define SIZE_MESSAGE_PING 4
+
+const char* g_program_name = "Ping";
+
+// TODO consider rename to: prepare_for_bind_client
+static int
+prepare_for_bind(struct sockaddr_un* addr_ptr)
+{
+    int32_t sock_path_max_len;
+    int32_t ret;
+
+    char error_message[SIZE_ERROR_MESSAGE];
+    char* sock_path;
+
+    ret = 0;
+    sock_path = 0;
+
+    sock_path_max_len = sizeof(addr_ptr->sun_path) - 1;
+    sock_path = (char*) calloc(sock_path_max_len, sizeof(char));
+    if (0 == sock_path) {
+        snprintf(
+                error_message,
+                SIZE_ERROR_MESSAGE,
+                "Failed to allocate memory for socket path");
+        print_error(error_message);
+        return -1;
+    }
+
+    memset(addr_ptr, 0, sizeof(struct sockaddr_un));
+
+    addr_ptr->sun_family = AF_UNIX;
+
+    // Construct client socket path with the help of PID
+    snprintf(
+            sock_path,
+            sock_path_max_len,
+            "/tmp/%s.%ld",
+            g_program_name,
+            (long) getpid());
+
+    strncpy(addr_ptr->sun_path, sock_path, sock_path_max_len);
+
+#ifndef NDEBUG
+    (void) printf("Constructed socket path: %s\n", addr_ptr->sun_path);
+#endif // NDEBUG
+
+    free(sock_path);
+    sock_path = 0;
+
+    return 0;
+}
+
+static void
+construct_server_address(struct sockaddr_un* addr_ptr)
+{
+    memset(addr_ptr, 0, sizeof(struct sockaddr_un));
+    addr_ptr->sun_family = AF_UNIX;
+    strncpy(
+            addr_ptr->sun_path,
+            SOCK_PATH_SERVER,
+            sizeof(addr_ptr->sun_path) - 1);
+}
+
+static int
+send_and_echo_messages(
+        int32_t const * const sock_fd_ptr,
+        struct sockaddr_un* address_server_ptr)
+{
+    char message[SIZE_MESSAGE_PING];
+    int32_t ret;
+
+    ret = 0;
+    memset(message, 0, SIZE_MESSAGE_PING);
+
+    strncpy(message, "Ping", SIZE_MESSAGE_PING);
+
+    ret = sendto(
+            *(sock_fd_ptr),
+            message,
+            SIZE_MESSAGE_PING,
+            0,
```

Comment:
Please use `flags` variable instead of magic 0.

Reply:
Done in: https://github.com/arbiter0xf/neotetris/pull/8/commits/c8d23f3e2e9b407564156deb3e69af11b93af847